### PR TITLE
feat: add sylaps into the app-store in video category

### PR DESCRIPTION
**Pull request by hthetiot**
## What does this PR do?

Add Sylaps (aka https://Sylaps.com) video conference service to the Apps.

Screenshot 2023-01-26 at 14 20 02

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] Visit Apps
- [x] Install Sylaps
- [x] Create a event
- [x] Set location to Sylaps
- [x] Test the generated sylaps link

Changes by @hariombalhara 
- imageSrc was still being used for Location select images. Replaced that with logo.
- Ensured that appMetaData.ts module itself derives `dirName` from `slug`. 
- Ensured that converting logo to absolute URL happens at a single place which is `appMetaDeta.ts`
- Dynamic Video apps need not have video in their slug name now. 
- Removed _metadata.ts and added config.json. _metadata.ts is deprecated

### CODE REVIEW

**Comment by hthetiot** [comment]
I think codacy failure is wrong, i have checked other DESCRIPTION files they use 2 spaces.
Also they say, "remove 2 spaces" and "add 2 spaces" on the same line, so false positive i guess.

Screenshot 2023-01-26 at 14 55 11

**Comment by hthetiot** [comment]
I might need so also implement this changes https://github.com/calcom/cal.com/pull/6712/files

File: packages/app-store/sylaps/_metadata.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export const metadata = {
+  name: "Sylaps",
+  description: _package.description,
+  installed: true,
+  type: "sylaps_video",
+  imageSrc: "/api/app-store/sylaps/icon.svg",
+  variant: "conferencing",
+  categories: ["video"],
+  logo: "/api/app-store/sylaps/icon.svg",
+  publisher: "Cal.com",
```
Comment (PeerRich) [inline packages/app-store/sylaps/_metadata.ts:14]:
```suggestion
  publisher: "hthetiot",
```

Comment (PeerRich) [reply packages/app-store/sylaps/_metadata.ts:14]: OR "Sylaps" if you belong to them

Comment (hthetiot) [reply packages/app-store/sylaps/_metadata.ts]: Yes I'm Sylaps founder.
Updated to published to Sylaps Inc.
Thx for review.

**Comment by hthetiot** [comment]
merged main to fix CI

File: packages/app-store/sylaps/_metadata.ts (hunk 1 above)
Comment (PeerRich) [reply packages/app-store/sylaps/_metadata.ts]: rest looks great! will try today. just made a sylaps account to tet

**Comment by PeerRich** [comment]
bear with me, our CI is currently borked, just switched to a new Github workflows. I think your PR is fine, I just wanna wait for the green check marks to come back

**Review by zomars — DISMISSED** [review]
Just tested this locally and stumbled upon this case.

The success page says: "Sylaps: Meeting url is in the confirmation email"

image

But the the confirmation email just says "Sylaps":

image

No link is there whatsoever

**Comment by hariombalhara** [comment]
Have fixed the issue reported by @zomars locally, but can't push the fix. @hthetiot can you allow edits on the PR from maintainers. See https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md#:~:text=Be%20sure%20to%20check%20the%20%22Allow%20edits%20from%20maintainers%22%20option%20while%20creating%20you%20PR.

**Comment by hthetiot** [comment]
> Have fixed the issue reported by @zomars locally, but can't push the fix. @hthetiot can you allow edits on the PR from maintainers. See https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md#:~:text=Be%20sure%20to%20check%20the%20%22Allow%20edits%20from%20maintainers%22%20option%20while%20creating%20you%20PR.
> 
> The changes that I have done, I have already added in PR description.

Will do, thx, keep you posted.

**Comment by PeerRich** [comment]
hey @hthetiot were the changes applied? happy to merge this week

**Comment by hthetiot** [comment]
> hey @hthetiot were the changes applied? happy to merge this week

Sorry need more time, I will attempt to finish PR next week.

**Comment by hthetiot** [comment]
Sorry it can only be done on initial PR creation:
"You can set commit permissions when you first create a pull request from a fork."

@zomars Can you share with me the fix instead ? Or if you prefer i can re-create the PR.

**Comment by hthetiot** [comment]
@zomars Should work as expected now.

Screenshot 2023-04-02 at 20 54 24

**Comment by hthetiot** [comment]
Test video (Recorded with Sylaps)
[Recording 2023-04-02T19-10_04.webm](https://user-images.githubusercontent.com/8635/229373894-f3eea948-1d1e-4140-8f2b-63f5269bfb7f.webm)

File: packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { PartialReference } from "@calcom/types/EventManager";
+import type { VideoApiAdapter, VideoCallData } from "@calcom/types/VideoApiAdapter";
+
+const SylapsApiAdapter = (): VideoApiAdapter => {
+  return {
+    getAvailability: () => {
+      return Promise.resolve([]);
+    },
+    createMeeting: async (event: CalendarEvent): Promise<VideoCallData> => {
```
Comment (hariombalhara) [inline packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts:12]:
```suggestion
    createMeeting: async (): Promise<VideoCallData> => {
```

File: packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      const meetingID = uuidv4();
+      return Promise.resolve({
+        type: "sylaps_video",
+        id: meetingID,
+        password: "",
+        url: "https://sylaps.com/r/" + meetingID,
+      });
+    },
+    deleteMeeting: async (uid: string): Promise<void> => {
```
Comment (hariombalhara) [inline packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts:21]:
```suggestion
    deleteMeeting: async (): Promise<void> => {
```

File: packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      Promise.resolve();
+    },
+    updateMeeting: (bookingRef: PartialReference, event: CalendarEvent): Promise<VideoCallData> => {
```
Comment (hariombalhara) [inline packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts:24]:
```suggestion
    updateMeeting: (bookingRef: PartialReference): Promise<VideoCallData> => {
```

**Review by hariombalhara — COMMENTED** [review]
Looks good to merge. There are a few things that I would like to change as I can't push to the branch still, I would merge and then fix.
- Linting warnings
- InstallAppButton.tsx file isn't needed now.
-

**Comment by hthetiot** [comment]
@hariombalhara

> Linting warnings
It complain anout space here:

But other file are the same:

> InstallAppButton.tsx file isn't needed now.

Ok will delete then.

File: packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts [hunk 5]
```diff
@@ -21,7 +21,7 @@ const SylapsApiAdapter = (): VideoApiAdapter => {
     deleteMeeting: async (uid: string): Promise<void> => {
       Promise.resolve();
     },
-    updateMeeting: (bookingRef: PartialReference, event: CalendarEvent): Promise<VideoCallData> => {
+    updateMeeting: (bookingRef: PartialReference): Promise<VideoCallData> => {
```
Comment (hthetiot) [inline packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts:24]: @hariombalhara The warning you fixed i think was fixed by 046d5e952339f37c6193154c741da7ea66a41cf0
But if you dont like unsued i understant.

**Comment by hthetiot** [comment]
> InstallAppButton.tsx file isn't needed now.

Does it mean in the future if we want to connect Sylaps Account it can be used?
If so for now i d'like to leave it, alternativly let me know and i will delete, or pls feel free tp suggest like you did on lint.

File: packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts (hunk 5 above)
Comment (hthetiot) [reply packages/app-store/sylapsvideo/lib/VideoApiAdapter.ts:24]: We may want to alto remove CalendarEvent import i guess.

**Comment by hariombalhara** [comment]
>> InstallAppButton.tsx file isn't needed now.
> 
> Does it mean in the future if we want to connect Sylaps Account it can be used? If so for now i d'like to leave it, alternatively let me know and i will delete, or pls feel free tp suggest like you did on lint.

Actually, having or not having that file makes no difference functionally. Every App automatically uses a common InstallAppButton component. It is a legacy file(wouldn't have come if you used App Store Cli to create the app) which probably came because you copied it from an old app(pre AppStore Cli)

**Review by hariombalhara — APPROVED** [review]

**Comment by hthetiot** [comment]
> InstallAppButton.tsx file isn't needed now.
deleted

**Comment by hthetiot** [comment]
Thx @hariombalhara for merge.
Nice CI and environnement you have for this project. 
Thx for open sourcing.

**Comment by hariombalhara** [comment]
@hthetiot There is an issue at the moment in enabling the app. I would let you know once it resolves

**Comment by hthetiot** [comment]
> @hthetiot There is an issue at the moment in enabling the app. I would let you know once it resolves

@hariombalhara 
I can see the sylaps app now thank a lot. But I did found a bug doh, if you visit the store listing after installing an app (sylaps here), it still show install button on the main store page after a search, if you click install you will have a javascript error about startWith and redirect to a cal.com/undefined. I will create an issue with video and reproducible steps.

**Pull request merged**